src/server/zset_family.cc

```diff
@@ -1762,8 +1762,10 @@ void ZBooleanOperation(CmdArgList args, ConnectionContext* cntx, bool is_union,
   for (auto& op_res : maps) {
     if (op_res.status() == OpStatus::SKIPPED)
       continue;
-    if (!op_res)
+    if (!op_res) {
+      cntx->transaction->Conclude();
```

Comment:
Please note that we conclude if store is false - see the comment at line 1758.  
so maybe
```suggestion
      if (store)
        cntx->transaction->Conclude();

```

Reply:
Sure updated